src/main/java/org/folio/inventory/consortium/consumers/ConsortiumInstanceSharingHandler.java

```diff
@@ -67,121 +94,489 @@ public Future<String> handle(KafkaConsumerRecord<String, String> event) {
+            .onComplete(ignored -> {
+              String message = format("Instance with InstanceId=%s has been shared to the target tenant=%s", instanceId, targetTenant);
+              sendCompleteEventToKafka(sharingInstanceMetadata, COMPLETE, message, kafkaHeaders);
+              promise.complete(message);
             }).onFailure(throwable -> {
-              String errorMessage = format("Error sharing Instance with InstanceId=%s to the target tenant %s.",
-                instanceId, targetTenant);
+              String errorMessage = format("Error updating Instance with InstanceId=%s on source tenant=%s.", instanceId, sourceTenant);
               sendErrorResponseAndPrintLogMessage(errorMessage, sharingInstanceMetadata, kafkaHeaders);
+              promise.fail(throwable);
             });
-        } else if ("MARC".equals(srcInstance.getSource())) {
-          String errorMessage = format("Error sharing Instance with InstanceId=%s and source=MARC to the target tenant %s. " +
-            "Not implemented yet.", instanceId, targetTenant);
+        }, throwable -> {
+          String errorMessage = format("Error sharing Instance with InstanceId=%s to the target tenant=%s.", instanceId, targetTenant);
           sendErrorResponseAndPrintLogMessage(errorMessage, sharingInstanceMetadata, kafkaHeaders);
-          return Future.failedFuture(errorMessage);
+          promise.fail(throwable);
+          return null;
+        });
+    } catch (Exception ex) {
+      LOGGER.error(format("Failed to sharing instance with ImportId=%s to target tenant=%s",
+        sharingInstanceMetadata.getInstanceIdentifier(), sharingInstanceMetadata.getTargetTenantId()), ex);
+      promise.fail(ex);
+    }
+    return promise.future();
+  }
+
+  private Future<String> publishInstanceWithMarcSource(Record marcRecord, SharingInstance sharingInstanceMetadata, Map<String, String> kafkaHeaders) {
+
+    String instanceId = sharingInstanceMetadata.getInstanceIdentifier().toString();
+    LOGGER.info("publishInstanceWithMarcSource:: Importing MARC record for instance with InstanceId={} to target tenant={}.",
+      sharingInstanceMetadata.getInstanceIdentifier(), sharingInstanceMetadata.getTargetTenantId());
+
+    Promise<String> promise = Promise.promise();
+    ChangeManagerClient targetManagerClient = getChangeManagerClient(kafkaHeaders);
+
+    initJobExecution(targetManagerClient, kafkaHeaders)
+      .compose(jobExecution -> {
+        String jobExecutionId = jobExecution.getString(ID);
+        LOGGER.info("New JobExcution initialized for import MARC record for instance with InstanceId={}, " +
+          "jobExecutionId={}", instanceId, jobExecutionId);
+
+        return setDefaultJobProfileToJobExecution(jobExecutionId, targetManagerClient)
+          .compose(ignore -> {
+            Object parsedRecord = JsonObject.mapFrom(marcRecord.getParsedRecord().getContent());
+
+            return postRecordToParsing(jobExecutionId, true,
```

Comment:
i could not determine what this method does but its name. Please add more detail.